src/main/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStep.java

```diff
@@ -91,16 +93,21 @@
             Job<?,?> job = build.getParent();
 
             Run<?,?> previousRun = build.getPreviousCompletedBuild();
-            JobPropertyTrackerAction previousAction = null;
+            JobPropertyTrackerAction previousAction = job.getAction(JobPropertyTrackerAction.class);
             boolean previousHadStep = false;
-            if (previousRun != null) {
-                previousAction = previousRun.getAction(JobPropertyTrackerAction.class);
-
+            if (previousAction == null && previousRun != null) {
```

Comment:
Nit: no need to call `getPreviousCompletedBuild` unless `previousAction == null`.